common/cards/card-plugins/hermits/goodtimeswithscar-rare.js

```diff
@@ -26,106 +26,58 @@ class GoodTimesWithScarRareHermitCard extends HermitCard {
-			const coinFlip = flipCoin(currentPlayer)
-			currentPlayer.custom[attacker.hermitCard.cardInstance] = coinFlip
-			currentPlayer.custom[this.id] = attacker.hermitCard.cardInstance
-
-			return target
-		})
-
-		// next turn attack on scar
-		game.hooks.attack.tap(this.id, (target) => {
-			const {opponentPlayer} = game.ds
-			const instance = opponentPlayer.custom[this.id]
-			if (!instance) return target
-
-			// Check that we are attacking card that used the Deathloop ability
-			if (target.row.hermitCard.cardInstance !== instance) return target
-
-			const coinFlip = opponentPlayer.custom[instance]
-
-			if (!coinFlip || coinFlip[0] === 'tails') return target
-
-			target.recovery.push({amount: this.recoverAmount})
-			return target
-		})
-
-		// After attack check if scar's ability was used
-		game.hooks.attackResult.tap(this.id, (target, turnAction, attackState) => {
-			const {currentPlayer, opponentPlayer} = game.ds
-
-			const instance = opponentPlayer.custom[this.id]
-			if (!instance) return target
-			if (target.row.hermitCard.cardInstance !== instance) return target
-			const coinFlip = opponentPlayer.custom[instance]
-			if (!coinFlip) return target
-			if (!target.died && !target.revived) return target
-
-			currentPlayer.coinFlips[this.id] = coinFlip
-			if (!target.revived) delete opponentPlayer.custom[instance]
-			delete opponentPlayer.custom[this.id]
-		})
-
-		// ailment death
-		game.hooks.hermitDeath.tap(this.id, (recovery, deathInfo) => {
-			const {playerState, row} = deathInfo
-			if (row.hermitCard.cardId !== this.id) return recovery
-			const instance = playerState.custom[this.id]
-			if (!instance) return recovery
-			const coinFlip = playerState.custom[instance]
-			if (!coinFlip) return recovery
-
-			playerState.coinFlips[this.id] = coinFlip
-
-			if (coinFlip[0] === 'heads') {
-				recovery.push({amount: this.recoverAmount})
-			} else {
-				delete playerState.custom[instance]
-			}
-			delete playerState.custom[this.id]
-
-			return recovery
-		})
-
-		// If scar did not revive by his next turn delete the flag
-		game.hooks.turnStart.tap(this.id, () => {
-			const {currentPlayer} = game.ds
-			if (currentPlayer.custom[this.id]) {
-				const instance = currentPlayer.custom[this.id]
-				delete currentPlayer.custom[this.id]
-				delete currentPlayer.custom[instance]
+	onAttach(game, instance, pos) {
+		const {player, otherPlayer, row, rowIndex} = pos
+		const reviveNextTurn = this.getInstanceKey(instance, 'reviveNextTurn')
+		const scarRevivedKey = this.getInstanceKey(instance, 'revived')
+
+		player.hooks.onAttack[instance] = (attack) => {
+			if (
+				attack.id !== this.getInstanceKey(instance) ||
+				attack.type !== 'secondary'
+			)
+				return
+
+			player.custom[reviveNextTurn] = true
+		}
+
+		otherPlayer.hooks.afterAttack[instance] = () => {
+			if (player.board.activeRow !== rowIndex) return
+			if (player.custom[reviveNextTurn] && !player.custom[scarRevivedKey]) {
+				if (row?.health !== 0) return
```

Comment:
It's worth checking to see that a) it's not undefined and b) it's less than or equal to 0